### PR TITLE
refactoring-advice-1704

### PR DESCRIPTION
**Pull request by Markiyan000**

### CODE REVIEW

**Review by forestik — APPROVED** [review]

File: core/src/main/java/greencity/controller/AdviceController.java [hunk 1]
```diff
@@ -79,6 +78,23 @@ public LanguageTranslationDTO getRandomAdviceByHabitIdAndLanguage(
...
+     * @return instance of {@link AdviceVO}
+     * @author Markiyan Derevetskyi
+     * */
+    @ApiOperation("Get advice by id")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = HttpStatuses.OK),
+            @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @GetMapping("/{adviceId}")
```
Comment (vterlyha) [inline core/src/main/java/greencity/controller/AdviceController.java:93]: /{adviceId} -> /{id}. It is better by conventions. You already know this is advice controller, so no need to add advice prefix

File: dao/src/main/java/greencity/entity/Advice.java [hunk 2]
```diff
@@ -18,9 +18,32 @@
...
+    /**
+     * Method that removes AdviceTranslation from Advice and also sets up a two-way relationship.
+     *
+     * @param adviceTranslation {@link AdviceTranslation}
+     * @author Markiyan Derevetskyi
+     */
+    public void removeAdviceTranslation(AdviceTranslation adviceTranslation) {
+        translations.remove(adviceTranslation);
+        adviceTranslation.setAdvice(null);
+    }
```
Comment (vterlyha) [inline dao/src/main/java/greencity/entity/Advice.java:48]: Entity should not have business logic

File: dao/src/test/java/greencity/repository/AdviceTranslationRepoTest.java [hunk 3]
```diff
@@ -23,6 +30,35 @@
...
+
+    private AdviceTranslation thirdAdviceTranslation = AdviceTranslation.builder()
+            .id(3L)
+            .language(Language.builder().id(3L).code("ru").build())
+            .content("Привет")
+            .advice(Advice.builder().id(3L)
+                    .habit(Habit.builder().id(3L).image("image_three.png")
+                            .build())
+                    .build()).build();
+
```
Comment (vterlyha) [inline dao/src/test/java/greencity/repository/AdviceTranslationRepoTest.java:61]: Please move these objects creation to ModelUtils class and use them as local variables.

**Review by Yura09 — APPROVED** [review]

File: core/src/main/java/greencity/controller/AdviceController.java (hunk 1 above)
Comment (Markiyan000) [reply core/src/main/java/greencity/controller/AdviceController.java:93]: Changed

File: dao/src/main/java/greencity/entity/Advice.java (hunk 2 above)
Comment (Markiyan000) [reply dao/src/main/java/greencity/entity/Advice.java:48]: Removed methods.

File: dao/src/test/java/greencity/repository/AdviceTranslationRepoTest.java (hunk 3 above)
Comment (Markiyan000) [reply dao/src/test/java/greencity/repository/AdviceTranslationRepoTest.java:61]: Moved to ModelUtils.

**Review by IvanBegar — APPROVED** [review]

**Pull request merged**